mxcube3/app.py

```diff
@@ -2,288 +2,400 @@
 Module that contains application wide settings and state as well as functions
 for accessing and manipulating those.
 """
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-
 import datetime
 import os
 import sys
 import logging
+import traceback
+import pickle
 import atexit
 import json
 
 from logging import StreamHandler, NullHandler
 from logging.handlers import TimedRotatingFileHandler
 
+from mxcubecore import HardwareRepository as hwr
 from mxcubecore import removeLoggingHandlers
-import queue_entry
-
-from mxcube3 import blcontrol
+from mxcubecore.HardwareObjects import (QueueManager, queue_entry)
 
-# Below variables used for internal application state
+from mxcube3.video import streaming_processes
+from mxcube3.logging_handler import MX3LoggingHandler
 
-# SampleID and sample data of currently mounted sample
-CURRENTLY_MOUNTED_SAMPLE = ""
 
-# Sample location of sample that are in process of being mounted
-SAMPLE_TO_BE_MOUNTED = ""
+class MXCUBECore():
+    # The HardwareRepository object
+    HWR = None
 
-# Method used for sample centring
-CENTRING_METHOD = queue_entry.CENTRING_METHOD.LOOP
+    # Below, all the HardwareObjects made available through this module,
+    # Initialized by the init function
 
-# Look up table for finding the limsID for a corresponding queueID (QueueNode)
-NODE_ID_TO_LIMS_ID = {}
+    # BeamlineSetup
+    beamline = None
+    # XMLRPCServer
+    actions = None
+    # Plotting
+    plotting = None
 
-# Initial file list for user, initialized at login, for creating automatic
-# run numbers
-INITIAL_FILE_LIST = []
+    @staticmethod
```

Comment:
why static method? (as opposed to adding the initialization in the __init__)

Reply:
I wanted singleton like behavior for these classes (But I'm not certain we actually need a singleton). So I created a static class for this an the variables that were previously global are this class variables and all methods static (as there are no instance)